libs/datasets/sources/state_provided_testing_dataset.py

```diff
@@ -0,0 +1,40 @@
+import dataclasses
+from libs.datasets import data_source
+from datapublic.common_fields import CommonFields
+from libs.datasets.sources import can_scraper_helpers as ccd_helpers
+from libs.datasets.timeseries import MultiRegionDataset
+
+
+class StateProvidedTestingDataset(data_source.CanScraperBase):
+    """Data source connecting to state health department sourced testing data."""
+
+    SOURCE_TYPE = "StateProvidedTesting"
+
+    VARIABLES = [
+        ccd_helpers.ScraperVariable(
+            variable_name="pcr_tests_positive",
+            measurement="cumulative",
+            provider="state",
+            unit="specimens",
+            common_field=CommonFields.POSITIVE_TESTS,
+        ),
+        ccd_helpers.ScraperVariable(
+            variable_name="pcr_tests_total",
+            measurement="cumulative",
+            provider="state",
+            unit="specimens",
+            common_field=CommonFields.TOTAL_TESTS,
+        ),
+    ]
+
+    def make_dataset(self) -> MultiRegionDataset:
+        dataset = super().make_dataset()
+        ts = dataset.timeseries_bucketed
+        # Calculate 7-day rolling sum to match CDC test positivity convention.
+        # See "Test Percent Positivity Metric" at
+        # https://data.cdc.gov/Public-Health-Surveillance/Weekly-COVID-19-County-Level-of-Community-Transmis/dt66-w6m6/
+        ts[CommonFields.TOTAL_TESTS_7D] = ts[CommonFields.TOTAL_TESTS].rolling(window=7).sum()
+        ts[CommonFields.POSITIVE_TESTS_7D] = ts[CommonFields.POSITIVE_TESTS].rolling(window=7).sum()
+        ts = ts.drop(columns=[CommonFields.TOTAL_TESTS, CommonFields.POSITIVE_TESTS])
```

Comment:
Oh, wow.  Nice work bending this to match the CDC methodology.  Arguably we could be doing this downstream in libs/metrics/test_positivity.py...  e.g. bake it into `DivisionMethod`.  But if this is already tested and you want to ship it as-is, I'm fine with that.

Reply:
I realized here that I'm taking a rolling sum of cumulative values which doesn't really make sense... So I need to either change this or use the new/daily data we pull from the scraper. In that process I'll look into doing that in the `DivisionMethod` create method if it's straightforward, but if not I might just opt to continue to do that here if we're okay with that

Reply:
Oh, right. Good catch.  Sounds good.

Reply:
changed this to use daily data in e9aee489a0cdb8f3b975364762431548b76f290b, and will sanity check the output in the AM

Reply:
Manually calculating the results for LA county with,

```python
   import pandas as pd
   # dataset source 
   df = pd.read_csv("https://data.chhs.ca.gov/dataset/f333528b-4d38-4814-bebb-12db1f10f535/resource/046cdd2b-31e5-4d34-9ed3-b48cdbc4be7a/download/covid19cases_test.csv")
   df = df[["date", "area", "positive_tests", "total_tests"]].set_index(["date", "area"]
   la = df.xs("Los Angeles", level="area")
   la["ptw"] = la["positive_tests"].rolling(window=7).sum()
   la["ttw"] = la["total_tests"].rolling(window=7).sum()
   la["ratio"] = la["ptw"] / la["ttw"]
```

results in: 
```python
la.dropna().tail(7)
```
```
            positive_tests  total_tests      ptw       ttw     ratio
date                                                                
2022-11-29          4672.0      44176.0  21337.0  186160.0  0.114616
2022-11-30          4325.0      37001.0  22707.0  195489.0  0.116155
2022-12-01          3686.0      31674.0  25587.0  219884.0  0.116366
2022-12-02          2917.0      24334.0  25706.0  224236.0  0.114638
2022-12-03           930.0       6883.0  24299.0  215845.0  0.112576
2022-12-04           165.0       1055.0  22351.0  201935.0  0.110684
2022-12-05            56.0        509.0  16751.0  145632.0  0.115023
```

where `ratio` is the test-positivity metric. This output matches the output of the local/test run, so I feel comfortable trusting these calculations. 

It does look like tests are reported by specimen collection date, so the most recent days have fewer observations (I imagine they get updated over time as more results come in) so this could mildly sway the metric, but I think that's okay.